### PR TITLE
Adds Service Virtualization section

### PR DESCRIPTION
**Pull request by lordofthejars**
Adds Service virtualization chapter to the docs, so you show some usage to cut service dependencies, so testing in production is easier.

I need to figure out how to apply the s2i approach in Hoverfly Docker image but overall is finished.

### CODE REVIEW

File: readme.adoc [hunk 1]
```diff
@@ -1145,6 +1145,193 @@ oc logs -f `oc get pods|grep recommendation-v2|awk '{ print $1 }'` -c recommenda
...
+We'll create version 2 of preferences service. 
+But in this case instead of communicating with recommendation service, we are goinng to communicate with a virtualized recommendation service.
+
+Service virtualization can be understood as something similar as mocking where instead of mocking components (classes), you are mocking remote services.
+
+For this concrete case, virtualized recommendation service will return `recommendation v2 from 'virtualized': 2`.
+
+=== Preference:v2
+
+Change `PreferencesController.java` like the following and creating a "v2" docker image.
```
Comment (rafabene) [inline readme.adoc]: a  'cd preference/java/springboot' should go here.

File: readme.adoc [hunk 2]
```diff
@@ -1145,6 +1145,193 @@ oc logs -f `oc get pods|grep recommendation-v2|awk '{ print $1 }'` -c recommenda
...
+
+The "v2" tag during the Docker build is significant.
+
+There is also a second `deployment.yml` file to label things correctly
+
+=== Docker build (if you have access to Docker daemon)
+
+[source, bash]
+----
+cd preference/java/springboot
```
Comment (rafabene) [inline readme.adoc]: Move "cd" before editing the file.

File: readme.adoc [hunk 3]
```diff
@@ -1145,6 +1145,193 @@ oc logs -f `oc get pods|grep recommendation-v2|awk '{ print $1 }'` -c recommenda
...
+IMPORTANT: We have a 2nd Deployment to manage the v2 version of preference.
+
+[source, bash]
+----
+oc apply -f <(istioctl kube-inject -f ../../kubernetes/Deployment-v2.yml) -n tutorial
+
+oc get pods -w
+----
+
+`cd ../../..`
```
Comment (rafabene) [inline readme.adoc]: Fix the formating

File: readme.adoc [hunk 4]
```diff
@@ -1145,6 +1145,193 @@ oc logs -f `oc get pods|grep recommendation-v2|awk '{ print $1 }'` -c recommenda
...
+
+[source, bash]
+----
+cd recommendation/virtualized
+
+docker build -t example/recommendation:virtualized .
+docker images | grep recommendation
+
+oc apply -f <(istioctl kube-inject -f ../kubernetes/Deployment-virtualized.yml) -n tutorial
+oc get pods
```
Comment (rafabene) [inline readme.adoc]: oc get pods -w

**Comment by rafabene** [comment]
Let's move this to ROOT/advanced/readme_virtualization.adoc

I'll adjust the main readme.adoc after. You don't need to handle it.

**Comment by lordofthejars** [comment]
@rafabene Updated.

**Pull request merged**